### PR TITLE
Fix &convert with vectors.

### LINKED ISSUE #368: &convert cannot deduce type when working on list
While `&convert` can be used to transform scalar values with user-provided functions, it fails to deduce the correct destination type for lists as of fa3e0b22160fa059092782fe3586e2482942dc06.
```
$ cat /tmp/bla.spicy
module Foo;

function fn1(x: uint8): uint8 { return x; }

function fn2(xs: vector<uint8>): vector<uint8> { return xs; }

public type P = unit {
    x: uint8 &convert=fn1($$);
    xs: uint8[] &size=1 &convert=fn2($$); # Fails to deduce.
};
```

```
$ spicyc -j /tmp/bla.spicy
[error] /tmp/bla.spicy:9:9: type '<unknown type>' cannot be used for function parameter
[error] spicyc: aborting after errors
```

### PR DESCRIPTION
**Pull request by rsmmr**
Fixes #368.

### CODE REVIEW

File: hilti/include/rt/types/vector.h [hunk 1]
```diff
@@ -233,8 +233,8 @@ class Vector : protected std::vector<T, Allocator> {
 
     Vector(std::initializer_list<T> init, const Allocator& alloc = Allocator()) : V(std::move(init), alloc) {}
 
-    Vector(const rt::List<T>& l) : std::vector<T>(l.begin(), l.end()) {}
-    Vector(rt::List<T>&& l) : std::vector<T>(std::move_iterator(l.begin()), std::move_iterator(l.end())) {}
+    Vector(const rt::List<T>& l) : std::vector<T, Allocator>(l.begin(), l.end()) {}
+    Vector(rt::List<T>&& l) : std::vector<T, Allocator>(std::move_iterator(l.begin()), std::move_iterator(l.end())) {}
```
Comment (bbannier) [inline hilti/include/rt/types/vector.h:237]:
```suggestion
    Vector(const rt::List<T>& l) : V(l.begin(), l.end()) {}
    Vector(rt::List<T>&& l) : V(std::move_iterator(l.begin()), std::move_iterator(l.end())) {}
```

File: hilti/src/ast/node_ref.cc [hunk 2]
```diff
@@ -11,7 +11,7 @@ NodeRef::NodeRef(Node& n) : _control(n._control()) {}
 
 Node* NodeRef::_node() const {
     if ( ! _control )
-        throw node_ref::Invalid("accesd to uninitalized node reference");
+        throw node_ref::Invalid("access to uninitialized node reference");
```
Comment (bbannier) [inline hilti/src/ast/node_ref.cc:14]: Not added here, but we should eventually make `node_ref::Invalid` a proper `RuntimeError`.

File: spicy/include/ast/types/unit-items/field.h [hunk 3]
```diff
@@ -103,6 +103,21 @@ class Field : public hilti::NodeBase, public spicy::trait::isUnitItem {
     // Node interface.
     auto properties() const { return node::Properties{{"engine", to_string(_engine)}}; }
 
+    // Helper function for vector fields that returns the type of the
+    // vector's elements. The helper can be used in situations where the
+    // field type might not be fully resolved it. It computes the type
```
Comment (bbannier) [inline spicy/include/ast/types/unit-items/field.h:108]:
```suggestion
    // field type might not be fully resolved. It computes the type
```

File: hilti/src/ast/node_ref.cc (hunk 2 above)
Comment (rsmmr) [reply hilti/src/ast/node_ref.cc:14]: It's a purely compiler/AST-side error, so not really a runtime error I would say.

**Pull request merged**